common/persistence/cassandra/cassandraVisibilityPersistence.go

```diff
@@ -187,6 +187,19 @@ type (
 	}
 )
 
+func (v *cassandraVisibilityPersistence) Query(
+	request *p.InternalListWorkflowExecutionsRequest,
+	template string,
+	params ...interface{},
+) gocql.Query {
+	if request.IsCron != nil {
+		template += `AND is_cron = ? ALLOW FILTERING `
```

Comment:
I don't think we want to use `allow filtering`?

Reply:
If I don't put ALLOW FILTERING, cassandra complains since this is not an indexed column. Is there another way to do this?

Reply:
Yeah, that's why I mentioned a secondary index is needed if we want to support it. We can sync during standup but my current understanding is we don't have to support it for cassandra and sql, just like we don't support querying based on tasklist when using those two DB for visibility.

Reply:
Non-indexed query will not scale well in Cassandra(same as MySQL/Postgres, but they don't complains).

What is the background of this? I wonder why not just implement this in ESVisibility only? We have decided to move to ES for new requirements long time ago as database doesn't have good support for searching like ElasticSearch.

Reply:
A little more some background, even secondary index will not perform/scale well in production. @vancexu did a comprehensive scalability test 1-2 yrs ago, then we decided to add Sampling to C* visibility https://github.com/uber/cadence/blob/master/common/persistence/visibilitySamplingClient.go 
And later on we moved to ES based visibility. 

ES visibility should be default for Cadence. You may change the internal wiki that es based docker compose is the default so that internal developer can use it by default. I am refactor visibility in https://github.com/uber/cadence/pull/4241 so that other databases will not have to implement visibility.